internal/cmd/connect/install_utils.go

```diff
@@ -0,0 +1,439 @@
+			list.Add(&listOut{
+				Number:      strconv.Itoa(i + 1),
+				Path:        installation.Location.Path,
+				Description: installation.Use,
+			})
+		}
+
+		listStr, err := list.PrintString()
+		if err != nil {
+			return nil, err
+		}
+
+		promptMsg := "\nThe plugin can be installed in any of the following Confluent Platform installations. Enter the number corresponding to the installation you would like to use:\n%sTo cancel, press Ctrl-C"
+		f := form.New(form.Field{
+			ID:     "installation",
+			Prompt: fmt.Sprintf(promptMsg, listStr),
+			Regex:  `^\d$`,
+		})
+		if err := f.Prompt(form.NewPrompt(os.Stdin)); err != nil {
+			return nil, err
+		}
+		choice, err := strconv.Atoi(f.Responses["installation"].(string))
+		if err != nil || choice < 1 || choice > len(installations) {
+			return nil, errors.Errorf("your choice must be in the range %d to %d (inclusive)", 1, len(installations))
+		}
+		return &installations[choice-1], nil
+	}
+}
+
+func findInstallationDirectories() ([]platformInstallation, error) {
+	// Check in descending order of precedence:
+	//   - $CONFLUENT_HOME
+	//   - current directory
+	//   - standard rpm/deb
+	//   - based on the client
+
+	var result []platformInstallation
+	hasPackageInstallation := utils.DoesPathExist(filepath.FromSlash("/usr/bin/connect-distributed"))
+
+	// $CONFLUENT_HOME
+	confluentHome := os.Getenv("CONFLUENT_HOME")
+	if confluentHome != "" && hasArchiveInstallation(confluentHome) {
+		installation := platformInstallation{
+			Location: platformLocation{
+				Type: "ARCHIVE",
+				Path: confluentHome,
+			},
+			Use: "$CONFLUENT_HOME",
+		}
+		result = append(result, installation)
+	}
+
+	// current directory
+	currentDirectory, err := os.Getwd()
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to determine current working directory")
+	}
+	if hasArchiveInstallation(currentDirectory) {
+		installation := platformInstallation{
+			Location: platformLocation{
+				Type: "ARCHIVE",
+				Path: currentDirectory,
+			},
+			Use: "Current Directory",
+		}
+		result = append(result, installation)
+	}
+
+	// standard rpm/deb
+	if hasPackageInstallation {
+		installation := platformInstallation{
+			Location: platformLocation{
+				Type: "PACKAGE",
+				Path: filepath.FromSlash("/"),
+			},
+			Use: "Installed RPM/DEB Package",
+		}
+		result = append(result, installation)
+	}
+
+	// based on the client
+	cliPath, err := os.Executable()
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to determine path to CLI")
+	}
+	cliDirectory := filepath.Dir(cliPath)
+	cliUse := "CLI Installation Directory"
+	if filepath.ToSlash(cliDirectory) == "/usr/bin" && hasPackageInstallation {
+		installation := platformInstallation{
+			Location: platformLocation{
+				Type: "PACKAGE",
+				Path: filepath.FromSlash("/"),
+			},
+			Use: cliUse,
+		}
+		result = append(result, installation)
+	} else if filepath.Base(cliDirectory) == "bin" && hasArchiveInstallation(filepath.Dir(cliDirectory)) {
```

Comment:
how is these 2 conditions (`usr/bin` and `bin`) determined? Could there be any chance that neither of these 2 conditions are true?

Reply:
`/usr/bin` is where the CLI is installed for package manager installations, and `bin` is where it's installed in archive installations.

The second one is also local, so if the CLI determines that it's in side a folder called `bin` it will check the parent directory of `bin` to see if it's a Confluent Platform installation.

Unless the user relocates `confluent` for whatever reason, these should not change.